enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,171 @@
+But update requests like `desiredUpdate` have different schema requirements than output like `desired` and `availableUpdates`.
+For example, [the `force` property][api-force] makes sense in the `desiredUpdate` context, but not in the `availableUpdates` contexts.
+And release metadata like landing page URIs and release metadata makes sense in the `desired` and `availableUpdates` contexts, but not in the `desiredUpdate` context.
+This enhancement adds a new `Release` type to be used in `status` properties so we can distinguish between input and output schema.
+
+## Motivation
+
+Cincinnati provides [metadata about releases][cincinnati-metadata], including (since [here][cincinnati-channel-metadata]).
+Exposing that information to ClusterVersion consumers allows them to use live metadata, instead of talking to Cincinnati directly or hard-coding expected values.
+For example, the console currently [hard-codes an expected set of available channels][console-channels] for a given release; with this enhancement they could look up available channels in `status.desired.metadata["io.openshift.upgrades.graph.release.channels"]`.
+
+### Goals
+
+* Exposing release metadata provided by upstream Cincinnati for the currently-desired and available-update releases.
+* Documenting some well-known metadata keys.
+
+### Non-Goals
+
+* Restricting metadata keys to those documented in this enhancement.
+    Cincinnati may continue to add and remove keys as they see fit as long as they provide at least the keys required by this enhancement.
+* Teaching Cincinnati to support requests without `channel` query parameters.
+    This might be useful for recovering from users specifying [`desiredUpdate`][api-desiredUpdate] that were not in their configured [`channel`][api-channel], but I am fine leaving recovery to the users.
+    They should have received sufficient information to adjust their channel from wherever they received the target release they are supplying.
+
+## Proposal
+
+As proposed in [this API pull request][api-pull-request], to add a new type:
+
+
+```go
+// Release represents an OpenShift release in the release graph.
+// +k8s:deepcopy-gen=true
+type Release struct {
+  // version is a semantic versioning identifying the update version. When this
+  // field is part of spec, version is optional if image is specified.
+  Version string `json:"version"`
+
+  // image is a container image location that contains the update. When this
+  // field is part of spec, image is optional if version is specified and the
+  // availableUpdates field contains a matching version.
+  Image string `json:"image"`
+
+  // metadata is additional metadata associated with the release, such
+  // as associated channels (under
+  // io.openshift.upgrades.graph.release.channels) or a landing page
+  // (under url).
+  //
+  // +optional
+  Metadata map[string]string `json:"metadata,omitempty"`
```

Comment:
would it be safer to define a clearer api here, like these are Cincinnati supported ones and they are explicit fields.. and here's all the additional stuff...?

Reply:
I’m not sure we can or should trust this info.  I might prefer we extract this on demand from available versions instead.

Reply:
> I’m not sure we can or should trust this info. I might prefer we extract this on demand from available versions instead.

This would mean "no channel metadata ever", which seems a bit harsh.  I'm fine documenting metadata as unsigned, maybe even calling it `UnsignedMetadata` or `ExternalMeradata` to make it clear the the data is not coming directly from the signed release images.

Reply:
Currently we are transparently passing on the label keys from cincinatti responses. I would like we define certain label keys as part of the API so that consumers can depend on those, and CVO should ensure transformations.

controlling these as part of our API means it's clusterversion API's responsibility to provide guarantees.

Reply:
Maybe name it AdditionalMetadata ?

Reply:
> Currently we are transparently passing on the label keys from cincinatti responses...

I think that's ok, but can we move that aspect of the discussion over into [this thread][1]?

[1]: https://github.com/openshift/enhancements/pull/123#discussion_r430734344

Reply:
> Maybe name it AdditionalMetadata ?

Having AdditionalMetadata without a Metadata seemed awkward to me, so 182442f -> ca79c67 adds both, and discusses the security trade-offs between them.  That's getting into very serious ground, so I'd appreciate extra review of the new godoc comments.

Reply:
And I've pushed 63c027a -> da64394 adding a new *Not distinguishing based on data source* section under *Alternatives* trying to summarize the discussion from this thread.